kong/globalpatches.lua

```diff
@@ -157,6 +166,24 @@ return function(options)
         end
         return keys
       end
+      function SharedDict:ttl(key)
+        local item = self.data[key]
+        if item == nil then
+          return nil, "not found"
+        else
+          local expire_at = item.info.expire_at
+          if expire_at == nil then
+            return 0
+          else
+            local remaining = expire_at - ngx.now()
+            if remaining < 0 then
+              return nil, "not found"
+            else
+              return remaining
```

Comment:
what will happen when `remaining == 0` ?  would it expire at the next cycle, or maybe "TTL == 0  ⇒ never expire" ?

please ensure the case is covered by a test